src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -129,27 +129,60 @@ public static CodeWriter WriteFieldDeclarations(this CodeWriter writer, IEnumera
             return writer.Line();
         }
 
-        public static IDisposable WriteMethodDeclaration(this CodeWriter writer, MethodSignatureBase methodBase, params string[] disabledWarnings)
+        private static CodeWriter WriteMethodModifiers(this CodeWriter writer, MethodSignatureBase methodBase, bool isAbstract)
         {
-            foreach (var disabledWarning in disabledWarnings)
-            {
-                writer.Line($"#pragma warning disable {disabledWarning}");
-            }
-
             writer
                 .AppendRawIf("public ", methodBase.Modifiers.HasFlag(Public))
                 .AppendRawIf("internal ", methodBase.Modifiers.HasFlag(Internal))
                 .AppendRawIf("protected ", methodBase.Modifiers.HasFlag(Protected))
                 .AppendRawIf("private ", methodBase.Modifiers.HasFlag(Private));
 
+            writer.AppendRawIf("abstract ", isAbstract);
 
-            if (methodBase is MethodSignature method)
-            {
+            if (methodBase is MethodSignature)
                 writer
                     .AppendRawIf("virtual ", methodBase.Modifiers.HasFlag(Virtual))
+                    .AppendRawIf("new ", methodBase.Modifiers.HasFlag(New))
+                    .AppendRawIf("override ", methodBase.Modifiers.HasFlag(Override))
                     .AppendRawIf("static ", methodBase.Modifiers.HasFlag(Static))
-                    .AppendRawIf("async ", methodBase.Modifiers.HasFlag(Async));
+                    .AppendRawIf("async ", methodBase.Modifiers.HasFlag(Async) && !isAbstract); // abstract methods cannot have async modifier
```

Comment:
This is not the place where it should be checked. `MethodSignature` instantiation must fail if method gets both modifiers